src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java

```diff
@@ -117,6 +118,12 @@ public void analyzeFilesOnUserAction(Collection<VirtualFile> files, AnActionEven
     }
   }
 
+  public void analyzeFileAndTrySelectHotspot(VirtualFile file, String securityHotspotKey) {
+    AnalysisCallback callback = new ShowSecurityHotspotCallable(project, onTheFlyFindingsHolder, securityHotspotKey);
+    var task = new Analysis(project, List.of(file), TriggerType.OPEN_SECURITY_HOTSPOT, true, callback);
+    TaskRunnerKt.startBackgroundTask(project, ANALYSIS_TASK_TITLE, task::run);
```

Comment:
I would have expected to have a modal task. As the user explicitly request to open the hotspot in the IDE, I think it's better to show him that something is going on

Reply:
I will change that in a later PR, I don't think it is blocker for the merge.